packages/puppeteer-core/src/cdp/EmulationManager.ts

```diff
@@ -253,6 +259,12 @@ export class EmulationManager {
     viewportState: ViewportState
   ): Promise<void> {
     if (!viewportState.viewport) {
```

Comment:
let's add an early return here if the state is not active and call .setState two times if the viewport is null? first to reset it for existing targets with active: true, and the second time with active: true to avoid running clearDeviceMetricsOverride and setTouchEmulationEnabled for new targets?